### PR TITLE
Review of DM-8156: add support for Slurm to allocateNodes.py	

### PR DESCRIPTION
**Pull request by hsinfang**

### CODE REVIEW

File: python/lsst/ctrl/execute/allocator.py [hunk 1]
```diff
@@ -86,11 +86,11 @@ def __init__(self, platform, opts, configFileName):
                 user_home = os.getenv('HOME')
 
         if user_name is None:
-            raise RuntimeError("error: %s does not specify user name for platform %s" %
```
Comment (hsinfang) [inline python/lsst/ctrl/execute/allocator.py:89]: About line 82: would you consider changing or adding `self.platform == "verification":` and similar placeholders for the verification cluster platform? Seems convenient for the users.

**Comment by hsinfang** [comment]
Before merging, may you please rewrite the commit message in the imperative tense and squash as you see fit? (e.g. "more refactoring" is not very clear what the commit is about) 

https://developer.lsst.io/processes/workflow.html#git-commit-message-best-practices

File: python/lsst/ctrl/execute/namedClassFactory.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010 LSST Corporation.
```
Comment (hsinfang) [inline python/lsst/ctrl/execute/namedClassFactory.py]: year needs update
(and in a few other files as well)

File: python/lsst/ctrl/execute/namedClassFactory.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ 
+from builtins import object
+ 
+class NamedClassFactory(object):
+    """Create a new "name" class object
+ 
+    Parameters
+    ----------
+    name : `str`
+        the fully qualified nname of an object
```
Comment (hsinfang) [inline python/lsst/ctrl/execute/namedClassFactory.py]: nname -> name

File: python/lsst/ctrl/execute/allocator.py [hunk 4]
```diff
@@ -139,37 +141,34 @@ def createUniqueIdentifier(self):
...
         data structures suitable for use by the TemplateWriter object.
         """
-        resolvedName = envString.resolve(name)
-        configuration = CondorConfig()
-        configuration.load(resolvedName)
-        self.defaults["LOCAL_SCRATCH"] = envString.resolve(configuration.platform.localScratch)
-
-    def loadPbs(self, name):
-        """Loads all values from configuration and command line overrides into
+        # 11/17/16 - srp XXX
```
Comment (hsinfang) [inline python/lsst/ctrl/execute/allocator.py]: Did you forget to delete this?

File: python/lsst/ctrl/execute/allocator.py (hunk 1 above)
Comment (hsinfang) [reply python/lsst/ctrl/execute/allocator.py:89]: Why deleting the first `%s`?

File: python/lsst/ctrl/execute/allocator.py (hunk 4 above)
Comment (srp3rd) [reply python/lsst/ctrl/execute/allocator.py]: yes, I did forget.  Thanks

File: python/lsst/ctrl/execute/allocator.py (hunk 1 above)
Comment (srp3rd) [reply python/lsst/ctrl/execute/allocator.py:89]: The first %s was missing.  Added "==" per request.

File: python/lsst/ctrl/execute/namedClassFactory.py (hunk 2 above)
Comment (srp3rd) [reply python/lsst/ctrl/execute/namedClassFactory.py]: Updated in all files.

**Comment by srp3rd** [comment]
"Before merging, may you please rewrite the commit message in the imperative tense and squash as you see fit? (e.g. "more refactoring" is not very clear what the commit is about)"

I updated these, and fixed the copyright year.

**Pull request closed without merging**